libs/vkd3d/command.c

```diff
@@ -5068,6 +5346,44 @@ static void d3d12_command_list_clear(struct d3d12_command_list *list,
     }
 }
 
+static void d3d12_command_list_clear_attachment(struct d3d12_command_list *list, struct d3d12_resource *resource,
+        struct vkd3d_view *view, VkImageAspectFlags clear_aspects, const VkClearValue *clear_value, UINT rect_count,
+        const D3D12_RECT *rects)
+{
+    D3D12_RECT full_rect;
+    int attachment_idx;
+    bool full_clear;
+    unsigned int i;
+
+    /* If one of the clear rectangles covers the entire image, we
+     * may be able to use a fast path and re-initialize the image */
+    full_rect = d3d12_get_view_rect(resource, view);
+    full_clear = !rect_count;
+
+    for (i = 0; i < rect_count && !full_clear; i++)
+        full_clear |= !memcmp(&rects[i], &full_rect, sizeof(full_rect));
```

Comment:
Do we need to take scissor into account here?

Reply:
Viewport/Scissor doesn't affect clears, that was a D3D9 thing.

Reply:
And GL fwiw, but ye. As expected then.